content/articles/displaying-images-with-fresco-library-in-android/index.md

```diff
@@ -0,0 +1,221 @@
+            .setAutoPlayAnimations(true)
+            .build()
+```
+
+### Streaming JPEG images
+Streaming gives the capability to load an image in low resolution and improves it to high resolution as more of its contrast is loaded. Fresco gives a way to use streaming in Android by setting the method `setProgressiveRenderingEnabled()` to `true`.
+
+```kotlin
+val imageRequest= ImageRequestBuilder
+            .newBuilderWithSource(Uri.parse("https://image.ec21.com/image/algsorcings/oimg_GC09912344_CA09912371/Fresh-Cut-Flowers-From-Kenya.jpg"))
+            .setProgressiveRenderingEnabled(true)
+            .build()
+
+        binding.streamingimage.controller=Fresco.newDraweeControllerBuilder()
+            .setImageRequest(imageRequest)
+            .setOldController(binding.streamingimage.controller)
+            .build()
+```
+
+### Loading images from the internet 
+Loading images from the network is also a feature provided by fresco. This allows users to obtain the necessary image from the network directly. To have this capability, the method `setImageURI()` is used by passing the image link to the method.
+
+```kotlin
+binding.networkimage.setImageURI("https://upload.wikimedia.org/wikipedia/commons/7/7c/Mount_Kenya.jpg")
+```
+
+### Adding a circular shape to an image (using Kotlin)
+Without all the procedures of adding a dependency to enable the use of circular imageView, Fresco provides the ability to shape an image in a circular view by adding the following line of code.
+
+```kotlin
+binding.circularimage.setImageURI("https://www.worldatlas.com/r/w1200/upload/79/70/5a/shutterstock-549814942.jpg")
+```
+
+### Image Transformation
+Fresco makes use of the in-built library to perform different transformations for instance applying rotation and resizing.
+
+#### Rotation
+You can rotate images by specifying an angle of rotation in the image request, like so:
+
+`.setRotationOptions(RotationOptions.forceRotation(RotationOptions.ROTATE_90))`
+
+#### Resizing
+The main aim of resizing is to alter the image within the memory but does not necessarily change it. To resize, pass a `ResizeOptions` object when constructing an image request.
+
+`setResizeOptions(ResizeOptions(50,50))`
+
+### Adding a progressbar
+To have an idea if the image is loading, it is appropriate to add a progress bar that shows if the image is on a loading process. Add the following line of code in your `XML` layout. 
+
+```xml
+fresco:progressBarImage="@drawable/progress_bar"
+```
+
+When you run the app, you should expect to see the following output:
+
+![demo](/engineering-education/displaying-images-with-fresco-library-in-android/demo.gif)
+
+### Conclusion
+Fresco is a unique library that offers features such as animations, streaming, image loading, and transformation. Explore more on the other features provided by Fresco from their official documentation to help you in your android development process.
```

Comment:
Referring to the documentation for "other features" means that you didn't cover them in this article. Please give a none-contradicting conclusion.